health-checker/src/main/java/io/stargate/health/DataStoreHealthChecker.java

```diff
@@ -82,4 +102,45 @@ protected Result check() {
       return Result.unhealthy("Unable to access DataStore: " + e);
     }
   }
+
+  private void ensureTableExists(DataStore dataStore)
+      throws ExecutionException, InterruptedException {
+    // create a dedicated health-check keyspace to not risk conflicts with users keyspace
+    dataStore
+        .execute(
+            dataStore
+                .queryBuilder()
+                .create()
+                .keyspace(KEYSPACE_NAME)
+                .ifNotExists()
+                .withReplication(Replication.simpleStrategy(1))
+                .build()
+                .bind())
+        .get();
+
+    // create table
+    dataStore
+        .execute(
+            dataStore
+                .queryBuilder()
+                .create()
+                .table(KEYSPACE_NAME, TABLE_NAME)
+                .ifNotExists()
+                .column(EXPECTED_TABLE.columns())
+                .build()
+                .bind())
+        .get();
+
+    // insert record
+    dataStore
```

Comment:
Can we put this as part of the check so it's, insert record and then read it back?